src/main/java/uk/gov/justice/hmpps/prison/service/AdjudicationsService.java

```diff
@@ -412,6 +427,175 @@ public void deleteOicHearingResult(
         oicHearingResultRepository.delete(oicHearingResult);
     }
 
+    @Getter
+    @Setter
+    @AllArgsConstructor
+    class OicSanctionValidationResult {
+        Adjudication adjudication;
+        Long oicHearingId;
+        Long offenderBookId;
+    }
+
+    protected OicSanctionValidationResult validateOicSanction(Long adjudicationNumber) {
+        final var adjudication = adjudicationsRepository.findByParties_AdjudicationNumber(adjudicationNumber)
+            .orElseThrow(EntityNotFoundException.withMessage(format("Could not find adjudication number %d", adjudicationNumber)));
+
+        final var hearingResult = oicHearingResultRepository.findByAgencyIncidentIdAndFindingCode(adjudication.getAgencyIncidentId(), FindingCode.PROVED);
```

Comment:
Maybe this should be `oicHearingResultRepository.findOneByAgencyIncidentIdAndFindingCode` so it returns an Optional rather than a list?

Then you have a slight improvement in readability 

`oicHearingResultRepository.findByAgencyIncidentIdAndFindingCode(adjudication.getAgencyIncidentId(), FindingCode.PROVED).orElseThrow(..)`

Reply:
I afraid here can be more than 1 hearing result with 'proved' finding. If you are sure it always not more than 1 - I can change List to Optional.